Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm

```diff
@@ -247,7 +251,22 @@ static bool isFullWebBrowser(const String& bundleIdentifier)
 {
     ASSERT(!isInWebKitChildProcess());
 
+#if ENABLE(APP_BOUND_DOMAINS)
     static bool fullWebBrowser = WTF::processHasEntitlement("com.apple.developer.web-browser"_s);
+#elif PLATFORM(MAC)
+    static bool fullWebBrowser;
+    static std::once_flag once;
+    std::call_once(once, [] {
+        NSURL *currentURL = [[NSBundle mainBundle] bundleURL];
+        NSArray<NSURL *> *httpURLs = [[NSWorkspace sharedWorkspace] URLsForApplicationsToOpenURL:[NSURL URLWithString:@"http:"]];
+        bool canOpenHTTP = [httpURLs containsObject:currentURL];
+        NSArray<NSURL *> *httpsURLs = [[NSWorkspace sharedWorkspace] URLsForApplicationsToOpenURL:[NSURL URLWithString:@"https:"]];
+        bool canOpenHTTPS = [httpsURLs containsObject:currentURL];
+        fullWebBrowser = canOpenHTTPS && canOpenHTTP;
```

Comment:
This seems like a new definition, was it discussed somewhere?
Do we need this change in this patch, it seems TestWebKitAPI/WebKitTestRunner will anyway fall in the non browser cases?

Reply:
There is `return fullWebBrowser || isRunningTest(bundleIdentifier);` below, so we are treating test runners as browser.